### ISSUE TITLE
league-in-game is not working.

### ISSUE THREAD

**Issue by YI-YEE** [RCVolus/league-prod-toolkit#77]
In-game information Preview is not working.

All league-in-game settings (Item Display, Level Display) can be set and are reflected in the toolkit.

However, it is not reflected on the preview screen.

![image](https://user-images.githubusercontent.com/91971709/155855128-5490106c-d304-49ea-a7d5-e4ba4e4dc3e2.png)

**Comment by sdomi4** [comment]
These messages are often related to incomplete datadragon downloads. 
Can you check the following:
- plugin-static-league is shown as running / log messages indicate download is complete
- game assets are present in \league-prod-toolkit\modules\static-league\frontend\img\

if you see no log entries from plugin-static-league and the expected assets aren't present, you can force a redownload by clearing the last-downloaded-version in the config and relaunching the toolkit:

```
  "static-league": {
    "last-downloaded-version": ""
  },
```

**Comment by Laraakaa** [comment]
I believe that this could be related due to the league-ingame plugin not starting up correctly. With the small portion of logs you shared, it is not possible for us to pinpoint the exact issue.

Could you please share the full logs, from application start towards you getting the error (best to share in text form, feel free to redact whatever!)? Thank you very much in advace!

**Comment by YI-YEE** [comment]
```
> league-prod-toolkit-core@0.0.1 start
> npm run build && node --trace-warnings dist/core/app.js


> league-prod-toolkit-core@0.0.1 build
> tsc && webpack

asset main-bundle.js 29 KiB [compared for emit] (name: main)
asset rcv.png 26.1 KiB [compared for emit] [from: frontend/rcv.png] [copied]
asset frontend-lib.ts 6.54 KiB [compared for emit] [from: frontend/frontend-lib.ts] [copied]
asset style.css 310 bytes [compared for emit] [from: frontend/style.css] [copied]
./frontend/frontend-lib.ts 7 KiB [built] [code generated]
./core/eventbus/LPTE.ts 564 bytes [built] [code generated]
webpack 5.38.1 compiled successfully in 5440 ms

> league-prod-toolkit-core@0.0.1 postbuild
> copyfiles core/web/views/**.* core/web/public/**.* dist

2022-02-27T06:38:54.231Z [info ] main        :  _          _       _____           _ _    _ _
2022-02-27T06:38:54.233Z [info ] main        : | |    ___ | |     |_   _|__   ___ | | | _(_) |_
2022-02-27T06:38:54.234Z [info ] main        : | |   / _ \| |       | |/ _ \ / _ \| | |/ / | __|
2022-02-27T06:38:54.235Z [info ] main        : | |__| (_) | |___    | | (_) | (_) | |   <| | |_
2022-02-27T06:38:54.236Z [info ] main        : |_____\___/|_____|   |_|\___/ \___/|_|_|\_\_|\__|
2022-02-27T06:38:54.237Z [info ] main        :
2022-02-27T06:38:54.237Z [info ] lpte-svc    : Initialized event bus.
2022-02-27T06:38:54.240Z [info ] module-svc  : Initializing module service.
2022-02-27T06:38:54.399Z [info ] module-svc  : Initialized 19 module(s). Now loading plugins.
2022-02-27T06:38:54.400Z [info ] module-svc  : Loaded 19 plugin(s).
2022-02-27T06:39:16.377Z [info ] main        : LoL Toolkit started up successfully.
2022-02-27T06:39:16.383Z [info ] server      : Listening for requests on http://localhost:3003
2022-02-27T06:39:16.390Z [info ] module-svc  : Plugin status changed: plugin=layout-volu-europe, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.391Z [info ] module-svc  : Plugin status changed: plugin=config, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.399Z [info ] module-svc  : Plugin status changed: plugin=example-module, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.400Z [info ] module-svc  : Plugin status changed: plugin=league-end-of-game, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.403Z [info ] module-svc  : Plugin status changed: plugin=prod-clock, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.405Z [info ] module-svc  : Plugin status changed: plugin=provider-webapi, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.413Z [info ] module-svc  : Plugin status changed: plugin=rcv-caster, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.416Z [info ] module-svc  : Plugin status changed: plugin=rcv-gfx, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.417Z [info ] module-svc  : Plugin status changed: plugin=rcv-rune-gfx, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.418Z [info ] module-svc  : Plugin status changed: plugin=rcv-teams, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.421Z [info ] module-svc  : Plugin status changed: plugin=rcv-tournament-tree, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.425Z [info ] module-svc  : Plugin status changed: plugin=state-league, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.427Z [info ] module-svc  : Plugin status changed: plugin=valorant-pregame, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.429Z [info ] module-svc  : Plugin status changed: plugin=valorant-state, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.437Z [warn ] lpte-svc    : Awaiting event timed out. namespace=reply, type=request-bnm3fjhcsl04wiumf, timeout=5000
2022-02-27T06:39:16.438Z [error] lpte-svc    : Request timed out after 5000ms. Request meta={"channelType":"REQUEST","type":"request","namespace":"config","version":1,"sender":{"name":"database","version":"1.0.0","mode":"PLUGIN","path":"F:\\롤\\옵저버툴\\league-prod-toolkit-develop\\modules\\database"},"reply":"request-bnm3fjhcsl04wiumf"}
2022-02-27T06:39:16.441Z [warn ] plugin-database: database config could not be loaded
2022-02-27T06:39:16.444Z [warn ] lpte-svc    : Awaiting event timed out. namespace=reply, type=request-bnm3fjhcsl04wiut1, timeout=5000
2022-02-27T06:39:16.445Z [error] lpte-svc    : Request timed out after 5000ms. Request meta={"channelType":"REQUEST","type":"request","namespace":"config","version":1,"sender":{"name":"league-in-game","version":"1.0.0","mode":"PLUGIN","path":"F:\\롤\\옵저버툴\\league-prod-toolkit-develop\\modules\\league-in-game"},"reply":"request-bnm3fjhcsl04wiut1"}
2022-02-27T06:39:16.446Z [warn ] plugin-league-in-game: config could not be loaded
2022-02-27T06:39:16.457Z [info ] module-svc  : Plugin status changed: plugin=themeing, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:16.576Z [info ] plugin-static-league: start downloading dragontail.tgz
2022-02-27T06:39:16.578Z [info ] plugin-static-league: start downloading additional files
2022-02-27T06:39:17.376Z [info ] module-svc  : Plugin status changed: plugin=valorant-static, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:39:19.520Z [info ] plugin-static-league: finish downloading additional files
2022-02-27T06:41:28.092Z [warn ] lpte-svc    : Awaiting event timed out. namespace=lpt, type=ready, timeout=150000
2022-02-27T06:41:28.096Z [error] plugin-league-end-of-game: Uncaught error in league-end-of-game: request timed out
Error: request timed out
    at Timeout._onTimeout (F:\롤\옵저버툴\league-prod-toolkit-develop\dist\core\eventbus\LPTEService.js:71:24)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
2022-02-27T06:41:28.217Z [warn ] lpte-svc    : Awaiting event timed out. namespace=lpt, type=ready, timeout=150000
2022-02-27T06:41:28.218Z [error] plugin-prod-clock: Uncaught error in prod-clock: request timed out
Error: request timed out
    at Timeout._onTimeout (F:\롤\옵저버툴\league-prod-toolkit-develop\dist\core\eventbus\LPTEService.js:71:24)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
2022-02-27T06:41:41.163Z [warn ] lpte-svc    : Awaiting event timed out. namespace=lpt, type=ready, timeout=150000
2022-02-27T06:41:41.166Z [error] plugin-provider-webapi: Uncaught error in provider-webapi: request timed out
Error: request timed out
    at Timeout._onTimeout (F:\롤\옵저버툴\league-prod-toolkit-develop\dist\core\eventbus\LPTEService.js:71:24)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
2022-02-27T06:41:41.218Z [warn ] lpte-svc    : Awaiting event timed out. namespace=lpt, type=ready, timeout=150000
2022-02-27T06:41:41.219Z [error] plugin-rcv-caster: Uncaught error in rcv-caster: request timed out
Error: request timed out
    at Timeout._onTimeout (F:\롤\옵저버툴\league-prod-toolkit-develop\dist\core\eventbus\LPTEService.js:71:24)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
2022-02-27T06:41:41.542Z [warn ] lpte-svc    : Awaiting event timed out. namespace=lpt, type=ready, timeout=150000
2022-02-27T06:41:41.543Z [error] plugin-rcv-teams: Uncaught error in rcv-teams: request timed out
Error: request timed out
    at Timeout._onTimeout (F:\롤\옵저버툴\league-prod-toolkit-develop\dist\core\eventbus\LPTEService.js:71:24)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
2022-02-27T06:41:41.684Z [warn ] lpte-svc    : Awaiting event timed out. namespace=lpt, type=ready, timeout=150000
2022-02-27T06:41:41.685Z [error] plugin-state-league: Uncaught error in state-league: request timed out
Error: request timed out
    at Timeout._onTimeout (F:\롤\옵저버툴\league-prod-toolkit-develop\dist\core\eventbus\LPTEService.js:71:24)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
2022-02-27T06:41:46.527Z [warn ] lpte-svc    : Awaiting event timed out. namespace=lpt, type=ready, timeout=150000
2022-02-27T06:41:46.528Z [error] plugin-themeing: Uncaught error in themeing: request timed out
Error: request timed out
    at Timeout._onTimeout (F:\롤\옵저버툴\league-prod-toolkit-develop\dist\core\eventbus\LPTEService.js:71:24)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
2022-02-27T06:42:32.231Z [info ] plugin-static-league: finish downloading dragontail.tgz
2022-02-27T06:42:32.236Z [info ] plugin-static-league: start unpacking dragontail.tgz
2022-02-27T06:44:03.807Z [info ] plugin-static-league: finish unpacking dragontail.tgz
2022-02-27T06:44:03.912Z [info ] plugin-static-league: moving files to frontend
2022-02-27T06:44:16.493Z [info ] plugin-static-league: delete versioned folder
2022-02-27T06:44:16.495Z [info ] plugin-static-league: start downloading centered images
2022-02-27T06:44:16.514Z [info ] plugin-static-league: finish moving files to frontend
2022-02-27T06:44:16.657Z [info ] plugin-static-league: finish downloading centered images
(node:22492) [DEP0147] DeprecationWarning: In future versions of Node.js, fs.rmdir(path, { recursive: true }) will be removed. Use fs.rm(path, { recursive: true }) instead
    at emitRecursiveRmdirWarning (node:internal/fs/utils:816:13)
    at Object.rmdir (node:internal/fs/promises:542:5)
    at StaticData.removeVersionDir (F:\롤\옵저버툴\league-prod-toolkit-develop\modules\static-league\dist\StaticData.js:142:37)
    at StaticData.copyDDragonFiles (F:\롤\옵저버툴\league-prod-toolkit-develop\modules\static-league\dist\StaticData.js:136:14)
2022-02-27T06:44:17.924Z [info ] plugin-static-league: finish deleting versioned folder
2022-02-27T06:44:17.938Z [info ] module-svc  : Plugin status changed: plugin=static-league, old=UNAVAILABLE, new=RUNNING
2022-02-27T06:44:18.069Z [info ] plugin-config: config for static-league saved!
```

I checked the download.

Some Plugin was Unavailable, so I restarted the tool.

After the reboot, all plugin is now available, but league-in-game Preview and still doesn't working.

I turned on the observer tool.

![image](https://user-images.githubusercontent.com/91971709/155871798-04b798bb-09c7-44b5-b010-f628c10209b0.png)

![Desktop Screenshot 2022 02 27 - 16 01 52 61](https://user-images.githubusercontent.com/91971709/155872064-167c6851-4b34-4fb0-ada4-d0a515ac0ca3.png)

**Comment by sdomi4** [comment]
Thank you for the logs :) Based on the output, the dowload has successfully completed, the timeout errors *should* no longer occur on the next start of the toolkit.
Just to confirm: Do you have the [observer tool](https://github.com/RCVolus/league-observer-tool) connected and the ingame modules synced?

**Comment by YI-YEE** [comment]
> Thank you for the logs :) Based on the output, the dowload has successfully completed, the timeout errors _should_ no longer occur on the next start of the toolkit. Just to confirm: Do you have the [observer tool](https://github.com/RCVolus/league-observer-tool) connected and the ingame modules synced?

yesssss ofcourse

**Comment by sdomi4** [comment]
Based on the log outputs, it looks like there may be some issues with the config. Could you post the contents of your config (\league-prod-toolkit-develop\modules\config\config.json) with your API key redacted?

Out of curiosity, which game servers are you trying to run the toolkit with? We only test on EUW, so other servers might not work the way we expect it to (Garena is sometimes a bit weird for example).

**Comment by YI-YEE** [comment]
![image](https://user-images.githubusercontent.com/91971709/155983467-e301df41-4fc5-4969-b5f5-42c77b03f183.png)

**Comment by Himyu** [comment]
Ok since you are trying to access the Korean server which game client do you with the [league-observer-tool](https://github.com/RCVolus/league-observer-tool)? is there a different one?

**Comment by YI-YEE** [comment]
if u mean the version, 
Game Client Version: 12.4.423.2790
Game Version: 12.4.4232790+branch.releases-12-4.code.public.content.release
Observer-toll Version: 2.3.0

**Comment by sdomi4** [comment]
Are you still seeing these config errors when starting the toolkit?
```
[warn ] plugin-database: database config could not be loaded
[warn ] plugin-league-in-game: config could not be loaded
```

Additionally, could you please export and upload some game data? Just spectate a game for a few minutes (so some level ups/ item buys are included) with the observer tool synced and export the API data by clicking the file icon in the InGame section.

**Comment by YI-YEE** [comment]
![image](https://user-images.githubusercontent.com/91971709/156105790-668daeca-4668-4328-9909-96768767d695.png)

[InGame-data.txt](https://github.com/RCVolus/league-prod-toolkit/files/8158801/InGame-data.txt)
(InGame-data.json)

**Comment by Himyu** [comment]
> ![image](https://user-images.githubusercontent.com/91971709/155983467-e301df41-4fc5-4969-b5f5-42c77b03f183.png)

if you look into the modules folder under what files do you see and in which file did you save your config as you showed us here?

**Comment by YI-YEE** [comment]
![image](https://user-images.githubusercontent.com/91971709/156202074-dbac7f63-19b0-4d7d-b8aa-61679408acc3.png)

F:\롤\옵저버툴\league-prod-toolkit-develop\modules\config

**Comment by sdomi4** [comment]
could you also screenshot the contents of the config module specifically?

**Comment by YI-YEE** [comment]
![image](https://user-images.githubusercontent.com/91971709/156210163-fc1aa432-f2b0-4362-b424-bd77f98d83cd.png)

**Comment by sdomi4** [comment]
could you please upload a copy of your config.json so we can try to reproduce this? Please redact your API key before uploading.

**Comment by YI-YEE** [comment]
[config.zip](https://github.com/RCVolus/league-prod-toolkit/files/8165599/config.zip)
thanks for the help ;))

**Comment by sdomi4** [comment]
Unfortunately, we are not able to reproduce the issue, even with your config file :(

I was able to start the toolkit without errors with this environment:
- Windows 10 21H2
- Node.js 16.7
- MongoDB 5.0.3

Is there anything (significantly) different in your setup?

We are sort of running out of ideas... What happens when you run the toolkit as an admin?

**Comment by Laraakaa** [comment]
Hey there! To gain more information about the issue you're facing, could you please run the application with debug-logs enabled, and send us the entire log produced as file?

To enable debug logging, change the [start.bat](https://github.com/RCVolus/league-prod-toolkit/blob/develop/start.bat) to the following:
```powershell
@echo off

powershell -Command "& {npm start -- --loglevel debug}"
```
and restart the prod toolkit, using the start.bat file.

**Comment by YI-YEE** [comment]
> Unfortunately, we are not able to reproduce the issue, even with your config file :(
> 
> I was able to start the toolkit without errors with this environment:
> 
> * Windows 10 21H2
> * Node.js 16.7
> * MongoDB 5.0.3
> 
> Is there anything (significantly) different in your setup?
> 
> We are sort of running out of ideas... What happens when you run the toolkit as an admin?

![image](https://user-images.githubusercontent.com/91971709/156737530-dde4affb-3009-443a-ae87-81dc0ea58860.png)
There's a crash that leaves this message.

**Comment by sdomi4** [comment]
Ok, the admin thing doesnt work at all :')
Could you try starting it normally with the debug instructions by [Laraakaa](https://github.com/Laraakaa)?

**Comment by YI-YEE** [comment]
> Hey there! To gain more information about the issue you're facing, could you please run the application with debug-logs enabled, and send us the entire log produced as file?
> 
> To enable debug logging, change the [start.bat](https://github.com/RCVolus/league-prod-toolkit/blob/develop/start.bat) to the following:
> 
> ```powershell
> @echo off
> 
> powershell -Command "& {npm start -- --loglevel debug}"
> ```
> 
> and restart the prod toolkit, using the start.bat file.

[start.txt](https://github.com/RCVolus/league-prod-toolkit/files/8185097/start.txt)
[start(1).txt](https://github.com/RCVolus/league-prod-toolkit/files/8185176/start.1.txt)

**Comment by YI-YEE** [comment]
[start (in game).txt](https://github.com/RCVolus/league-prod-toolkit/files/8185381/start.in.game.txt)

**Comment by Himyu** [comment]
Ok, our current working theory is that this has something to do with some requests being sent too early. To test that could you please start only the prod tool without the browser pages or the observer tool open. and wait until the startup is fully completed. After that start your observer tool and the browser again.
If that indeed helps we will fix that asap but at least you have a workaround for now.
If that does not help we have to take other measures to fix this bug.

**Comment by YI-YEE** [comment]
I've been waiting for 10 hours, but still not working. Hahahaha. ;((

i will wait for the next update! thanks for the help :)

**Comment by sdomi4** [comment]
Hi YI-YEE

There have been some updates to the ingame module which *might* fix the issue you are seeing. The instructions to update the module are in #81 if you want to give it a try :)

**Comment by sdomi4** [comment]
different issue after update, outputs in #91

**Comment by sdomi4** [comment]
this should hopefully be resolved with release 0.9, if this persists on a fresh install please reopen the issue
